test/distributed/_composable/test_composability/test_2d_composability.py

```diff
@@ -2,30 +2,131 @@
 
 import copy
 import functools
+import io
+from copy import deepcopy
 from typing import List, Type
 
 import torch
+import torch.distributed as dist
 import torch.distributed.checkpoint as dcp
 import torch.nn as nn
+import torch.nn.functional as F
 from torch.distributed._composable import replicate
 from torch.distributed._composable.fsdp import CPUOffloadPolicy, fully_shard
-from torch.distributed._tensor import init_device_mesh
+from torch.distributed._tensor import (
+    DTensor,
+    DTensor as DT,
+    init_device_mesh,
+    Replicate,
+    Shard,
+)
 from torch.distributed._tensor.debug.comm_mode import CommDebugMode
 from torch.distributed.checkpoint.state_dict import (
     get_model_state_dict,
     get_optimizer_state_dict,
+    set_optimizer_state_dict,
 )
 from torch.distributed.device_mesh import DeviceMesh
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.distributed.fsdp._common_utils import (
+    _get_module_fsdp_state,
+    clean_tensor_name,
+)
+from torch.distributed.fsdp.fully_sharded_data_parallel import StateDictType
+from torch.distributed.tensor.parallel import (
+    ColwiseParallel,
+    parallelize_module,
+    RowwiseParallel,
+)
+from torch.distributed.tensor.parallel.ddp import _pre_dp_module_transform
+from torch.distributed.tensor.parallel.fsdp import DTensorExtensions
+from torch.distributed.tensor.parallel.input_reshard import input_reshard
+from torch.nn.parallel import DistributedDataParallel as DDP
 from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
 from torch.testing._internal.common_fsdp import FSDPTest, MLP, MLPStack
-from torch.testing._internal.common_utils import run_tests, skipIfRocm
+from torch.testing._internal.common_utils import (
+    instantiate_parametrized_tests,
+    parametrize,
+    run_tests,
+    skipIfRocm,
+)
 from torch.testing._internal.distributed._tensor.common_dtensor import (
+    DTensorTestBase,
+    MLPModule,
     ModelArgs,
     Transformer,
+    with_comms,
 )
 from torch.testing._internal.distributed.checkpoint_utils import with_temp_dir
 
 
+def init_model(device_type, model_parallel_size=2):
```

Comment:
I agree that this needs some cleanup. These probably should not be defined at global scope but rather within the test class that uses them (or something like that).

Reply:
Moved into the corresponding class